lib/user_space/user_x86_64_atomic.h

```diff
@@ -55,72 +54,84 @@ static inline int64_t m0_atomic64_get(const struct m0_atomic64 *a)
 	return a->a_value;
 }
 
-/**
- atomically increment counter
-
- @param a pointer to atomic counter
-
- @return none
- */
-static inline void m0_atomic64_inc(struct m0_atomic64 *a)
-{
-	asm volatile("lock incq %0"
-		     : "=m" (a->a_value)
-		     : "m" (a->a_value));
-}
-
-/**
- atomically decrement counter
-
- @param a pointer to atomic counter
-
- @return none
- */
-static inline void m0_atomic64_dec(struct m0_atomic64 *a)
-{
-	asm volatile("lock decq %0"
-		     : "=m" (a->a_value)
-		     : "m" (a->a_value));
-}
-
 /**
    Atomically adds given amount to a counter
  */
-static inline void m0_atomic64_add(struct m0_atomic64 *a, int64_t num)
-{
+static inline void m0_atomic64_add(struct m0_atomic64 *a, int64_t num)                    
+{                                                                       
+#ifdef CONFIG_X86_64
 	asm volatile("lock addq %1,%0"
-		     : "=m" (a->a_value)
-		     : "er" (num), "m" (a->a_value));
+				: "=m" (a->a_value)
+				: "er" (num), "m" (a->a_value));
+#else
+	long result;                                                    
+    unsigned long tmp;
+	asm volatile("// atomic64_add \n"			\
+			"       prfm    pstl1strm, %2\n"	\
+			"1:     ldxr    %0, %2\n"			\
+			"       add     %0, %0, %3\n"		\
+			"       stxr    %w1, %0, %2\n"      \
+			"       cbnz    %w1, 1b"			\
+        	: "=&r" (result), "=&r" (tmp), "+Q" (a->a_value)	\
+        	: "Ir" (num));
```

Comment:
The slash "\" should be aligned to the same position.

Reply:
Formatted, but plz look again, if it is not looking pleasing once again